### PR TITLE
ENH: add tolerance argument to .sel and .reindex

### PR DESCRIPTION
**Pull request by shoyer**
`.sel` and `.reindex` now support the ``tolerance`` argument for controlling nearest-neighbor selection:

    In [5]: array = xray.DataArray([1, 2, 3], dims='x')

    In [6]: array.reindex(x=[0.9, 1.5], method='nearest', tolerance=0.2)
    Out[6]:
    <xray.DataArray (x: 2)>
    array([  2.,  nan])
    Coordinates:
      * x        (x) float64 0.9 1.5

This feature requires pandas v0.17 or newer.

### CODE REVIEW

**Comment by shoyer** [comment]
@jhamman I think I want to merge this and then issue a new release. We are doing more harm in waiting for ReadTheDocs at this point. Better to get the fixes out even if we can't make the stable docs yet.

File: xray/core/indexing.py [hunk 1]
```diff
@@ -117,25 +117,24 @@ def _try_get_item(x):
...
-
-
-def convert_label_indexer(index, label, index_name='', method=None):
+def convert_label_indexer(index, label, index_name='', method=None,
+                          tolerance=None):
     """Given a pandas.Index (or xray.Coordinate) and labels (e.g., from
     __getitem__) for one dimension, return an indexer suitable for indexing an
     ndarray along that dimension
     """
+    # backwards compatibility for pandas<0.16 (method) or pandas<0.17
```
Comment (jhamman) [inline xray/core/indexing.py]: What error do we get if we use the `tolerance` keyword with pandas 0.16?  Do we need to catch that ourselves?

**Comment by jhamman** [comment]
I'm fine with merging this now.  I agree we need to get this release out, with or without read the docs support.

File: xray/core/indexing.py (hunk 1 above)
Comment (shoyer) [inline xray/core/indexing.py]: You should get a TypeError, since the argument does not exist. I'll double check, though.

**Pull request merged**